docs/api/server/Environment-Variables.md

```diff
@@ -2,6 +2,461 @@
+* [`ONE_CLIENT_REPORTING_URL`](#one_client_reporting_url)
+* [`ONE_CLIENT_ROOT_MODULE_NAME`](#one_client_root_module_name)
+* [`ONE_CONFIG_ENV`](#one_config_env)
+* [`ONE_ENABLE_POST_TO_MODULE_ROUTES`](#one_enable_post_to_module_routes)
+* [`ONE_MAP_POLLING_MAX`](#one_map_polling_max)
+* [`ONE_MAP_POLLING_MIN`](#one_map_polling_min)
+
+## `HOLOCRON_MODULE_MAP_URL`
+
+**Runs In**
+* ✅ Production
+* ✅ Development
+
+URL where the [Module Map] is hosted.
+
+Defaults to `http://localhost:3001/static/module-map.json` if `NODE_ENV` is set to `development` so that [One App Dev CDN] can be leveraged for local development.
+
+**Shape**
+```bash
+HOLOCRON_MODULE_MAP_URL=String
+```
+
+**Example**
+```bash
+HOLOCRON_MODULE_MAP_URL=https://my-cdn.com/module-map.json
+```
+
+## `HOLOCRON_SERVER_MAX_MODULES_RETRY`
+
+**Runs In**
+* ✅ Production
+* ✅ Development
+
+Maximum amount of times One App should retry on failed module fetches. Defaults to `3`.
+
+**Shape**
+```bash
+HOLOCRON_SERVER_MAX_MODULES_RETRY=Integer
+```
+
+**Example**
+```bash
+HOLOCRON_SERVER_MAX_MODULES_RETRY=10
+```
+
+## `HOLOCRON_SERVER_MAX_SIM_MODULES_FETCH`
+
+**Runs In**
+* ✅ Production
+* ✅ Development
+
+Maximum number of [Holocron Modules] One App will try to load at a time when new Modules are found in [Module Map].
+
+Useful to configure in case of a large module map and/or a bad network connection. Defaults to `30`.
+
+**Shape**
+```bash
+HOLOCRON_SERVER_MAX_SIM_MODULES_FETCH=Integer
+```
+
+**Example**
+```bash
+HOLOCRON_SERVER_MAX_SIM_MODULES_FETCH=10
+```
+
+## `HTTPS_PORT`
+
+> ⚠️ Requires [`HTTPS_PRIVATE_KEY_PATH`] and [`HTTPS_PUBLIC_CERT_CHAIN_PATH`] to be set.
+
+**Runs In**
+* ✅ Production 
+* ✅ Development
+
+Sets the port on which One App will listen for requests. It defaults to undefined.
+
+**Shape**
+```bash
+HTTPS_PORT=Integer
+```
+
+**Example**
+```bash
+HTTPS_PORT=443
+```
+
+## `HTTPS_PRIVATE_KEY_PASS_FILE_PATH`
+
+**Runs In**
+* ✅ Production 
+* ✅ Development
+
+The file path to a file containing a shared passphrase for single private key (See [Node documentation on `passphrase` for `tls.createSecureContext`](https://nodejs.org/api/tls.html#tls_tls_connect_options_callback)). 
+
+**Shape**
+```bash
+HTTPS_PRIVATE_KEY_PASS_FILE_PATH=String
+```
+
+**Example**
+```bash
+HTTPS_PRIVATE_KEY_PASS_FILE_PATH=./some-extra-certs.pem
+```
+
+## `HTTPS_PRIVATE_KEY_PATH`
+
+**Runs In**
+* ✅ Production 
+* ✅ Development
+
+The file path to the private key of an SSL Certificate.
+
+**Shape**
+```bash
+HTTPS_PRIVATE_KEY_PATH=String
+```
+
+**Example**
+```bash
+HTTPS_PRIVATE_KEY_PATH=./some-private-key.pem
+```
+
+## `HTTPS_PUBLIC_CERT_CHAIN_PATH`
+
+**Runs In**
+* ✅ Production 
+* ✅ Development
+
+The file path to the public key of an SSL Certificate.
+
+**Shape**
+```bash
+HTTPS_PUBLIC_CERT_CHAIN_PATH=String
+```
+
+**Example**
+```bash
+HTTPS_PUBLIC_CERT_CHAIN_PATH=./some-cert.pem
+```
+
+## `HTTPS_TRUSTED_CA_PATH`
```

Comment:
For each env var can we define if they are required and if so in what situations?

For example this env var I believe is only required if `HTTPS_PORT` is provided.

Reply:
Fixed in 3aa1d50